flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/api/FlinkOutputFieldsDeclarer.java

```diff
@@ -0,0 +1,173 @@
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * 
+	 * Currently, Flink only supports the default output stream. Thus, pareamter {@code streamId} must be equals to
+	 * {@link Utils#DEFAULT_STREAM_ID}.
+	 * 
+	 * @throws UnsupportedOperationException
+	 *             if {@code streamId} is not equal to {@link Utils#DEFAULT_STREAM_ID}
+	 */
+	@Override
+	public void declareStream(final String streamId, final Fields fields) {
+		this.declareStream(streamId, false, fields);
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * 
+	 * Currently, Flink only supports the default output stream. Thus, pareamter {@code streamId} must be equals to
+	 * {@link Utils#DEFAULT_STREAM_ID}. Furthermore, direct emit is no supported by Flink and parameter {@code direct}
+	 * must be {@code false}.
+	 * 
+	 * @throws UnsupportedOperationException
+	 *             if {@code streamId} is not equal to {@link Utils#DEFAULT_STREAM_ID} or {@code direct} is {@code true}
+	 */
+	@Override
+	public void declareStream(final String streamId, final boolean direct, final Fields fields) {
+		if(!Utils.DEFAULT_STREAM_ID.equals(streamId)) {
+			throw new UnsupportedOperationException("Currently, only the default output stream is supported by Flink");
+		}
+		if(direct) {
+			throw new UnsupportedOperationException("Direct emit is not supported by Flink");
+		}
+		
+		this.outputSchema = fields;
+	}
+	
+	/**
+	 * Returns {@link TypeInformation} for the declared output schema. If no or an empty output schema was declared,
+	 * {@code null} is returned.
+	 * 
+	 * @return output type information for the declared output schema; or {@code null} if no output schema was declared
+	 * 
+	 * @throws IllegalArgumentException
+	 *             if more then 25 attributes are declared
+	 */
+	public TypeInformation<?> getOutputType() throws IllegalArgumentException {
+		if((this.outputSchema == null) || (this.outputSchema.size() == 0)) {
+			return null;
+		}
+		
+		Tuple t;
+		final int numberOfAttributes = this.outputSchema.size();
+		
+		if(numberOfAttributes == 1) {
+			return TypeExtractor.getForClass(Object.class);
+		} else if(numberOfAttributes <= 25) {
+			try {
+				t = Tuple.getTupleClass(numberOfAttributes).newInstance();
+			} catch(final InstantiationException e) {
+				throw new RuntimeException(e);
+			} catch(final IllegalAccessException e) {
+				throw new RuntimeException(e);
+			}
+		} else {
+			throw new IllegalArgumentException("Flink supports only a maximum number of 25 attributes.");
+		}
+		
+		// TODO: declare only key fields as DummyComparable
+		for(int i = 0; i < numberOfAttributes; ++i) {
+			t.setField(new DummyComparable(), i);
+		}
+		
+		return TypeExtractor.getForObject(t);
+	}
+	
+	/**
+	 * {@link DummyComparable} is a {@link Comparable} helper class that is used to get the correct
+	 * {@link TypeInformation} from {@link TypeExtractor} within {@link #getOutputType()}. If key fields are not
+	 * comparable, Flink cannot use them and will throw an exception.
+	 * 
```

Comment:
As the comment says: "If key fields are NOT comparable, Flink cannot use them and will throw an exception."

Any suggestion for a better name?